### DISCUSSION TITLE
[C#] How to set the default Base URL ?

### DISCUSSION THREAD

**Discussion by maximeozenne** [scalar/scalar#3226] [Q&A] [answered]
Hello, I'm having an issue using Scalar in a .NET 9 project.

Scalar is configured as following :

```
app.MapScalarApiReference(options =>
{
    options
        .WithTitle("DummyService")
        .WithTheme(ScalarTheme.Kepler)
        .WithDarkMode(true)
        .WithDarkModeToggle(true)
        .WithForceThemeMode(ThemeMode.Dark)
        .WithSidebar(true)
        .WithDefaultHttpClient(ScalarTarget.CSharp, ScalarClient.HttpClient)
        ;
});
```

My API is using both REST endpoints and gRPC endpoints. So in my `appsettings.json`, I have a `Kestrel` entry :

```
"Kestrel": {
  "Endpoints": {
    "Grpc": {
      "Url": "https://localhost:5001",
      "Protocols": "Http2"
    },
    "Http": {
      "Url": "http://localhost:80"
    },
    "Https": {
      "Url": "https://localhost:443"
    }
  }
}
```

When I start the app, the `Base URL` is **always** the `gRPC` endpoint :

![CleanShot 2024-09-24 at 15 18 27](https://github.com/user-attachments/assets/0f769df2-ccb7-4911-918c-098825e61caa)

So I have to manually switch to https.

Even if I change the order of the endpoints entry in my `appsettings.json` the `Base URL` list order still the same.

How can I set the default `Base URL` please ?

**Comment by xC0dex** [comment, marked as answer]
Hey @maximeozenne,

The server URL is provided by the OpenApi generator implementation of Microsoft. [This code](https://github.com/dotnet/aspnetcore/blob/main/src/OpenApi/src/Services/OpenApiDocumentService.cs#L195) uses the `IServerAddressesFeature` to get the server address.
One option _could_ be that you implement a custom `OpenApiDocumentTransformer` to set a custom server URL in the generated document.

I notice that the [`server` configuration](https://github.com/scalar/scalar/blob/main/documentation/configuration.md#servers-server) is missing in our .NET implementation. I'll add this! I think this is the preferred solution you are looking for.

**Comment by xC0dex** [comment]
PR is ready https://github.com/scalar/scalar/pull/3246

**Reply by maximeozenne** [reply]
Thank you! I'll try it as fast as I can!

**Reply by maximeozenne** [reply]
Hello @xC0dex, it's me again haha
I think the PR is blocked because of an error during the CI/CD, for the moment I can't pull any newer version of the nuget package

**Reply by xC0dex** [reply]
Hey,

currently we don't have an automated workflow for the .NET package. This fix is currently not released. 👀 @marclave could you create a new release for `Scalar.AspNetCore`? 🚀

**Reply by xC0dex** [reply]
@maximeozenne Its released! Please give it a try 🥳.

**Reply by maximeozenne** [reply]
Hello @xC0dex! Unfortunately, I'm still seeing the 1.2.4 only :c no new version on my side, even with pre-version activated

**Reply by xC0dex** [reply]
Hm. Maybe something wrong with the local NuGet cache? It's published on [NuGet.org](https://www.nuget.org/packages/Scalar.AspNetCore#versions-body-tab).

**Reply by maximeozenne** [reply]
Thank you, it works !! We are using an Azure private feed which is fetching from NuGet.org, so maybe there is a caching system and we had to wait before seeing the new version.

I was able to pull the 1.2.5, and the `.AddServer()` did the job amazingly. Thank you very much!!
